enhancements/ignition-spec-dual-support.md

```diff
@@ -0,0 +1,447 @@
+only be required as part of the MCO.
+
+Note that there exists three types of spec 2 configs:
+ - Those that are directly translatable to spec 3. This is the case for all existing IPI configs.
+ - Those that we’re not 100% sure we can directly translate, but we can infer what the user is doing and do a translation 
+ - Those that we’re 100% sure we CANNOT translate directly, and requires user input for us to correctly translate
+
+During phase 1 we should attempt to translate on a best-effort basis. If the
+cluster can be directly translated to spec 3, we will do so and use the now
+v3 spec version config in the MCD. If not, we will support both, and alert
+the user that there are untranslatable configs.
+
+During phase two, we should fail updates unless the cluster is fully on spec
+3 config. This effectively means that UPI clusters are at risk when upgrading
+to an MCO with dual support. Mitigation methods are discussed below.
+
+For backwards support (spec 3 to spec 2) the configs are 100% translatable.
+
+
+### RHCOS
+
+Phase 1:
+
+No changes needed.
+
+Phase 2:
+
+The RHCOS bootimage needs to be updated to ignition package v2.0+ . Required
+dependencies are discussed below.
+
+
+### Installer
+
+Phase 1:
+
+No changes needed.
+
+Phase 2:
+
+The bootstrap ignition configs are updated to be spec 3. The stub ignition
+configs for master/worker nodes are updated to spec 3, and referece a different
+endpoint of the MCS which will serve ignition spec 3 configs. RHCOS images pinned
+by the installer will be updated in conjuction to ones with ignition v2 (spec3).
+All spec 2 references are stripped from the installer.
+
+
+### MCO
+
+The MCO and its subcomponents are the most affected by this change. The
+aforementioned spec translator will be housed in the MCO. This means that the
+MCO would need to simultaneously vendor both ignition v0 and v2, and translate
+existing between the spec versions as needed.
+
+Phase 1:
+
+The MCD has the capability to understand both spec 2 and spec 3 configs,
+and lay down files as needed.
+
+The MCC has the capability to translate spec 2 to spec 3 configs. If the
+translation is completely successful, it will create final rendered spec 3
+configs, and the MCD will be instructed to use the spec 3 configs to lay down
+necessary files. Otherwise the MCD will continue using the existing
+spec 2 configs, and apply new spec 2/spec 3 configs.
+
+The MCS served configs generated by the MCC will be downtranslated by the
+MCC to spec 2. The MCS is not updated to serve spec 3 configs yet.
+
+Phase 2:
+
+All machineconfigs are switched to spec 3. Spec 2 support will exist in the form
+of served ignition configs to new nodes joining the cluster that have ignition
+v0 in the bootimage. MCS will handle this with 2 endpoints.
+
+The MCO will flat out reject spec 2 configs, and refuse to upgrade clusters
+that have spec 2 bits.
+
+The MCO will also throw an alert upon: seeing a spec 2 machineconfig applied to
```

Comment:
I don’t think we want to bother with alerts here. We can just validate the MachineConfig before the CR is committed.